async/bin/thales.ml

```diff
@@ -0,0 +1,373 @@
+let invalid_arg fmt = Fmt.kstrf (fun e -> invalid_arg e) fmt
+let (<.>) f g = fun x -> f (g x)
+
+module U = Unix
+
+type cipher =
+  [ `AES_128_CBC_SHA
+  | `AES_128_CBC_SHA256
+  | `AES_128_CCM
+  | `AES_128_GCM_SHA256
+  | `AES_256_CBC_SHA
+  | `AES_256_CBC_SHA256
+  | `AES_256_CCM
+  | `AES_256_GCM_SHA384
+  | `_3DES_EDE_CBC_SHA
```

Comment:
I noticed this due to the odd type name, and then wondered if we shouldn't be disabling the 3DES ciphers by default due to their insecurity.  See [here](https://security.stackexchange.com/questions/103992/tls-rsa-with-3des-ede-cbc-sha-reported-as-112-bits) for a discussion about this particular cipher combination and its effective entropy.  If most clients in the wild support AES ciphers, then perhaps we can drop the older ones from async-tls.